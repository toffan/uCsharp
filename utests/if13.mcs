int main() {
    int c1 = 3;
    int c2 = 0;
    int c3 = 1;

    if (c1 || c2 || c3) {
        int a = 5;
    } else {
        int b = 2;
    }

    return 0;
}
