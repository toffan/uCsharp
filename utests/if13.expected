JUMP __main
_main:
    LOADL 3
    LOADL 0
    LOADL 1
    LOAD (1) 3[LB]
    LOAD (1) 4[LB]
    SUBR BOr
    LOAD (1) 5[LB]
    SUBR BOr
    JUMPIF (0) ELSE_0
    LOADL 5
    JUMP ENDIF_0
    ELSE_0:
    LOADL 2
    ENDIF_0:

    LOADL 0
    RETURN (1) 0
    RETURN (0) 0
__main:
    CALL (SB) _main
    HALT
